preupg/utils.py

```diff
@@ -411,27 +411,33 @@ def get_convertors():
         return settings.text_converters.keys()
 
     @staticmethod
-    def get_assessment_version(dir_name):
-        if PreupgHelper.get_prefix() == "preupgrade":
-            matched = re.search(r'\D+(\d*)_(\d+)', dir_name, re.I)
-            if matched:
-                return [matched.group(1), matched.group(2)]
-            else:
-                return None
-        else:
-            matched = re.search(r'\D+(\d*)_(\D*)(\d+)', dir_name, re.I)
-            if matched:
-                return [matched.group(1), matched.group(3)]
-            else:
-                return None
+    def get_assessment_version(dir_name, base_dir=settings.source_dir):
+        section_name = settings.section_name
+        properties_file = os.path.join(base_dir,
+                                       dir_name,
+                                       settings.properties_ini)
+        src_os_ver = ConfigHelper.get_preupg_config_file(properties_file,
+                                                         key=settings.src_version_key,
+                                                         section=section_name)
+        dest_os_ver = ConfigHelper.get_preupg_config_file(properties_file,
+                                                          key=settings.dest_version_key,
+                                                          section=section_name)
+        if src_os_ver is None or dest_os_ver is None:
+            return None
+        return [src_os_ver, dest_os_ver]
 
     @staticmethod
-    def get_valid_scenario(dir_name):
-        matched = [x for x in dir_name.split(os.path.sep) if re.match(r'\D+(\d*)_(\D*)(\d+)(-results)?$', x, re.I)]
-        if matched:
-            return matched[0]
-        else:
-            return None
+    def get_valid_scenario(scenario_path):
+        """
+        >>> get_valid_scenario('/dir1/dir2/file.xml')
+        'dir2'
+        >>> get_valid_scenario('/dir1/dir2/dir3/')
+        'dir3'
+        >>> get_valid_scenario('/dir1/dir2/dir3')
+        'dir2'
+        # doesn't check directory at the end of path, is considered as a file
+        """
+        return os.path.basename(os.path.split(scenario_path)[0])
```

Comment:
Use _os.path.dirname_ instead of _os.path.split(scenario_path)[0]_. It is equivalent and more clear.

Consider renaming the method to _get_module_set_dirname_. The current name does not hint much what it's supposed to do.